test/plugins/adonis.spec.js

```diff
@@ -0,0 +1,386 @@
+
+          Context.getter('response', function () {
+            const config = new Config()
+            config.set('app.http.jsonpCallback', 'callback')
+            return new Response(this.request, config)
+          }, true)
+
+          ioc.autoload(path.join(__dirname), 'App')
+        })
+      } else {
+        before(() => {
+          const framework = require(`${frameworkPath}@${version}`)
+
+          Server = framework.get('@adonisjs/framework/src/Server')
+          Route = framework.get('@adonisjs/framework/src/Route/Manager')
+          RouteStore = framework.get('@adonisjs/framework/src/Route/Store')
+          Request = framework.get('@adonisjs/framework/src/Request')
+          Response = framework.get('@adonisjs/framework/src/Response')
+          Context = framework.get('@adonisjs/framework/src/Context')
+          Exception = framework.get('@adonisjs/framework/src/Exception')
+
+          Context.getter('response', function () {
+            return new Response(this.req, this.res, new Config())
+          }, true)
+        })
+      }
+
+      before(() => {
+        Context.getter('request', function () {
+          return new Request(this.req, this.res, new Config())
+        }, true)
+
+        setupResolver()
+        agent.load(plugin, 'adonis').then(() => {
+        })
+      })
+
+      beforeEach(() => {
+        this.exception = Exception
+        this.server = new Server(Context, Route, new Logger(), this.exception)
+
+        getPort().then(newPort => {
+          port = newPort
+          server = this.server.listen('localhost', port)
+          server.on('listening', () => {
+            axios
+              .get(`http://localhost:${port}/api/user`)
+              .catch((err) => {})
+          })
+        })
+      })
+
+      after(() => {
+        return agent.close()
+      })
+
+      afterEach(() => {
+        RouteStore.clear()
+        ioc.restore()
+        this.exception.clear()
+        server.close()
+      })
+
+      it('should perform automatic instrumentation on routers', done => {
+        Route.get('/api/user', function ({ response }) {
+          response.send('GET')
+        })
+
+        agent.use(traces => {
+          const spans = traces[0]
+
+          expect(spans[0]).to.have.property('service', 'test')
+          expect(spans[0]).to.have.property('name', 'adonis.middleware')
+          expect(spans[0].meta).to.have.property('component', 'adonis')
+
+          expect(spans[1]).to.have.property('service', 'test')
+          expect(spans[1]).to.have.property('name', '/api/user')
+          expect(spans[1].meta).to.have.property('component', 'adonis')
+          expect(spans[1].meta).to.have.property('span.kind', 'server')
+          expect(spans[1].meta).to.have.property('http.url', `http://localhost:${port}/api/user`)
+          expect(spans[1].meta).to.have.property('http.method', 'GET')
+          expect(spans[1].meta).to.have.property('http.status_code', '200')
```

Comment:
Can you add parentage checks for all of these?
```expect(spans[0].context()._parentId.toString()).to.equal(spans[1].context().toSpanId())```

It could be a good time to add a test utility module loaded w/ https://github.com/signalfx/signalfx-nodejs-tracing/blob/master/test/setup/all.js

Reply:
Not sure what you want me to add to `setup/core.js`